src/plugins/vis_type_vega/public/expressions/helpers.ts

```diff
@@ -44,6 +56,10 @@ export const cleanString = (rawString: string): string => {
   return rawString.replaceAll('"', '');
 };
 
+export const calculateLegendOffset = (legendPosition: string): number =>
+  // 18 is the default offset as of vega lite 5
+  legendPosition === 'top' || legendPosition === 'bottom' ? 0 : 18;
```

Comment:
Why do we need to explicitly calculate this offset based on legend position?

Reply:
When the legend is at a vertical position (top or bottom), it shifts the chart canvas up or down to compensate, and can cause it to get cut off in the panel. I believe the root cause is that the vega-lite autosize has some issue of not compensating for this within the panel. So to fix this, i set the offset to 0 so the canvas doesn't move to guarantee it does not get cut off.

Reply:
The issue does not arise when the legends are horizontal (left/right), so I believe it is a vega-lite bug.

Reply:
added a comment about this

Reply:
Thanks - this is good for now, although we'll probably eventually want a more robust solution to vega responsive sizing in general. I appreciate all the research and improvements you've made to help us get this far.

Reply:
Definitely definitely agreed. Overall I'd say the existing integration of vega lite autosizing within the embeddable panels is a big mess. There's too many places that needs to be touched to get it to work properly - vega-lite spec, modifying compiled vega spec, modifying the vega view